Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx

```diff
@@ -40,36 +43,66 @@ const ContestFilters = () => {
     );
 
     const renderFilterItem = useCallback(
-        ({ id, name }: IFilter) => (
-            <Button
-              type={ButtonType.plain}
-              onClick={() => handleFilterClick(id)}
-              className={styles.btnSelectFilter}
-            >
-                <Label type={LabelType.plain}>
-                    {name}
-                </Label>
-            </Button>
-        ),
-        [ handleFilterClick ],
+        ({ id, name }: IFilter) => {
+            const filterIsSelected = filters.some(f => f.id === id);
+            const type = filterIsSelected ? ButtonType.primary : ButtonType.secondary;
+            const size = filterIsSelected ? ButtonSize.medium : ButtonSize.small;
+
+            return (
+                <Button
+                    type={type}
+                    onClick={() => handleFilterClick(id)}
+                    className={styles.btnSelectFilter}
+                    text={name}
+                    size={size}
+                />
+        )},
+        [ handleFilterClick, filters ],
+    );
+    
+    const toggleFiltersExpanded = useCallback(
+        () => {
+            setExpanded(!expanded);
+        },
+        [expanded],
+    );
```

Comment:
This seems wrong. Controlling the `expanded` state here, independently of the `expanded` state in `ExpandButton`.

If `ExpandButton`s `onClick` is instead `onExpandChanged(expand)`, this will become:
```
const toggleFiltersExpanded = useCallback(
   (expanded) => {
      setExpanded(expanded);
    },
    [],
);
```

This way, both states will be aligned